src/plugins/unified_histogram/public/layout/hooks/use_lens_suggestions.ts

```diff
@@ -50,6 +62,63 @@ export const useLensSuggestions = ({
   const currentSuggestion = originalSuggestion ?? suggestions.firstSuggestion;
   const suggestionDeps = useRef(getSuggestionDeps({ dataView, query, columns }));
 
+  const histogramSuggestion = useMemo(() => {
+    if (
+      !currentSuggestion &&
+      dataView.isTimeBased() &&
+      query &&
+      isOfAggregateQueryType(query) &&
+      getAggregateQueryMode(query) === 'esql' &&
+      timeRange
+    ) {
+      let queryHasTransformationalCommands = false;
+      if ('esql' in query) {
+        TRANSFORMATIONAL_COMMANDS.forEach((command: string) => {
+          if (query.esql.toLowerCase().includes(command)) {
+            queryHasTransformationalCommands = true;
+            return;
+          }
+        });
+      }
+
+      if (queryHasTransformationalCommands) return undefined;
+
+      const interval = computeInterval(timeRange, data);
+      const language = getAggregateQueryMode(query);
+      const histogramQuery = `${query[language]} | eval uniqueName = 1
+        | EVAL timestamp=DATE_TRUNC(${interval}, ${dataView.timeFieldName}) | stats rows = count(uniqueName) by timestamp | rename timestamp as \`${dataView.timeFieldName} every ${interval}\``;
+      const context = {
+        dataViewSpec: dataView?.toSpec(),
+        fieldName: '',
+        textBasedColumns: [
+          {
+            id: `${dataView.timeFieldName} every ${interval}`,
+            name: `${dataView.timeFieldName} every ${interval}`,
```

Comment:
This can't be translated unfortunately. It is not used as label on the Lens suggestion api. It is the variable generated from the query. We dont support custom labels at the moment in text based mode. It is def one of the enhancements we could do.